### ISSUE TITLE
Introduce build and coverage status

### ISSUE THREAD

**Issue by sergeyklay** [beanstalkd/beanstalkd#396]
Hello,

Just a proposal to introduce build badge (in the readme.md file) and [code coverage metric](https://codecov.io). Also could you please setup [Travis Cron Jobs](https://docs.travis-ci.com/user/cron-jobs/) so that beanstalkd will be tested with [various compilers](https://docs.travis-ci.com/user/languages/c/#choosing-compilers-to-test-against) and [multi OS matrix](https://docs.travis-ci.com/user/multi-os/#example-multi-os-build-matrix) for example every day.

P.S. I propose introduce benchmark as a build matrix as well.

**Comment by ysmolsky** [comment]
This issues is rather an umbrella for multiple features. We can discuss and track progress here.

**Comment by sergeyklay** [comment]
I would be happy to provide initial Travis CI configuration. Feel free to ping me any time.

**Comment by ysmolsky** [comment]
I would love to review your PR. Please, do not hesitate.

**Comment by ysmolsky** [comment]
@sergeyklay can you please look at this pull and figure out what is wrong? https://github.com/beanstalkd/beanstalkd/pull/420

I am not friends with coverage in C, I made this work locally through passing flags as parameters to make, but it does not work in travis.

**Comment by kr** [comment]
At first glance this looks like it might be a problem with whitespace and quoting in the shell.

```
make: test-coverage': No such file or directory
```

It looks like the actual value of `$COVERAGE` is

```
LDFLAGS='-fprofile-arcs -ftest-coverage' CFLAGS='-fprofile-arcs -ftest-coverage'
```

And when `$COVERAGE` appears in the make command line it expands to 4 separate args: `LDFLAGS='-fprofile-arcs`, `-ftest-coverage'`, `CFLAGS='-fprofile-arcs`, and `-ftest-coverage'`.

Putting double quotes around `$COVERAGE` in the make command might help.

```
make $MAKEJOBS "$COVERAGE" || ...
```

This would turn the value of that variable, including any whitespace, into a single arg.

**Comment by kr** [comment]
On second glance, a single arg won't work either, since `$COVERAGE` is trying to set two parameters.

Maybe consider changing `CFLAGS=...` in Makefile to `CFLAGS?=...` (and same for LDFLAGS) and then the travis config can control those via environment variables instead of arguments to make.

**Comment by ysmolsky** [comment]
Thanks @kr. It was not correct approach to the `make` tool. @sergeyklay proposed better approach with `CFLAGS+=...` in PR #421.

**Comment by sergeyklay** [comment]
Closing, this is implemented. If there are bugs or I have made an oversight, please do get back in touch. Thanks

**Comment by ysmolsky** [comment]
Hmm, this should have been closed automatically. Checking why it has not happened.
I see that the message was this:
```
enable code coverage on Travis CI

* Enable code coverage on Travis CI
* Collect coverage reports only on Linux with GCC
* Cleaned up Travis CI config

Closes: #420, #396
```
If it was this:
```
Closes #420 
Closes #396
```
then it would have worked. It should be a new line for every mentioned ticket.

**Comment by sergeyklay** [comment]
I think message should be:
```diff
- Closes: #420, #396
+ Closes: #420
+ Closes: #396 
```

See: https://stackoverflow.com/questions/3547445/closing-multiple-issues-in-github-with-a-commit-message

**Comment by ysmolsky** [comment]
Yes, exactly. Also there is no need to put colon after the verb:
`Closes #XXX`
But I like this form even better: `Fixes #XXX`.
